### PR TITLE
Activate Github actions

### PR DESCRIPTION
**Pull request by mer-a-o**
### Description
This PR adds a yaml file for github actions with simple instructions to start a cycle on a self-hosted runner (in this case Cheyenne) with `run-test` label. The main goal here is to add this yaml and expand it in the future.

### CODE REVIEW

File: .github/workflows/cycle.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - name: checkout
+        uses: actions/checkout@v3
+
+      - name: run-mpas-workflow
+        run: |
+          source /etc/profile.d/modules.sh
+          source ~/.bashrc
+          export JEDI_OPT=/glade/work/jedipara/cheyenne/opt/modules
+          module use $JEDI_OPT/modulefiles/core
+          module load jedi/gnu-openmpi
```
Comment (jjguerrette) [inline .github/workflows/cycle.yml]: Are these environment settings needed for some aspect on your end of the testing?

They are not typically needed to initiate an MPAS-Workflow scenario suite.  The compiler/run environment is loaded individually for each cylc task in its job script (e.g., config/environmentForJedi.csh).

File: .github/workflows/cycle.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    types: [labeled]
+    branches: [develop]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  Run-mpas-workflow:
+    if: ${{ github.event.label.name == 'run-cycle' }}
```
Comment (jjguerrette) [inline .github/workflows/cycle.yml]: Do you mind changing `run-cycle` to `run-test` to match the name of `test.csh`? EDIT: or to match `test.yaml`, I suppose.

File: .github/workflows/cycle.yml (hunk 1 above)
Comment (mer-a-o) [reply .github/workflows/cycle.yml]: If they are not needed we can remove them. We only need to keep `source /etc/profile.d/modules.sh` if we have `module load` commands in the code.

File: .github/workflows/cycle.yml (hunk 2 above)
Comment (mer-a-o) [reply .github/workflows/cycle.yml]: sure

File: .github/workflows/cycle.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# This is a basic workflow to help you get started with Actions
+
+name: test-mpas-workflow
+
+# Controls when the workflow will run
+on:
+  # uncomment if we decide to run tests when merging to develeop branch
```
Comment (jjguerrette) [inline .github/workflows/cycle.yml]: `develeop` -> `develop`

**Review by jjguerrette — APPROVED** [review]

**Comment by jjguerrette** [comment]
Oh maybe rename `cycle.yml` to `test.yml` also.

**Review by jjguerrette — APPROVED** [review]

**Comment by jjguerrette** [comment]
Depending on what you envision going in the `workflows` directory in the future, I might propose that it be renamed to `gh-actions` or something similar in a follow-on PR.  Looks great to merge this one.

**Comment by jjguerrette** [comment]
Disregard my last comment.  I just noticed `workflows` is already under `.github`.  That will work great.

**Comment by jjguerrette** [comment]
@mer-a-o, is this ready to merge?

**Comment by mer-a-o** [comment]
@jjguerrette yes, it's ready.

**Pull request merged**